app/models/trip.rb

```diff
@@ -0,0 +1,120 @@
+require 'pry'
+
+class Trip < ActiveRecord::Base
+  validates :duration_in_seconds, numericality: { only_integer: true }
+  validates :start_date,          presence: true
+  validates :end_date,            presence: true
+  validates :start_station,       presence: true
+  validates :end_station,         presence: true
+  # validates :bike_id,             presence: true
+  validates :subscription_id,     presence: true
+  validates :zipcode_id,          presence: true
+
+  belongs_to :start_station, :class_name => "Station", :foreign_key => "start_station_id"
+  belongs_to :end_station,   :class_name => "Station", :foreign_key => "end_station_id"
+  belongs_to :subscription
+  belongs_to :zipcode
+
+  def self.total
+    Trip.count
+  end
+
+  def self.average_duration_of_ride
+    self.average("duration_in_seconds").to_f.round(3)
+  end
+
+  def self.longest_ride
+    self.maximum("duration_in_seconds")
+  end
+
+  def self.shortest_ride
+    self.minimum("duration_in_seconds")
+  end
+
+  def self.most_rides_starting
+    most = self.select("start_station_id").group("start_station_id").count
+    station = Station.find(most.invert.values.max)
+    "#{station.name} = #{most.values.max}"
+  end
+
+  def self.most_rides_ending
+    most = self.select("end_station_id").group("end_station_id").count
+    station = Station.find(most.invert.values.max)
```

Comment:
we're doing this in more than one place. can we breakdown this method into smaller, reusable chunks? 

also - again, i'd pull out the formatting here on the next line so that this method returns an object rather than data